src/uproot/extras.py

```diff
@@ -160,70 +160,55 @@ def xrootd_version():
             return None
 
 
-def lz4_block():
+def zlib():
     """
-    Imports and returns ``lz4``.
-
-    Attempts to import ``xxhash`` as well.
+    Import and return ``zlib``, or ``isal_zlib``, if available.
     """
     try:
-        import lz4.block
-        import xxhash  # noqa: F401
-    except ModuleNotFoundError as err:
-        raise ModuleNotFoundError(
-            """install the 'lz4' and `xxhash` packages with:
-
-    pip install lz4 xxhash
-
-or
-
-    conda install lz4 python-xxhash"""
-        ) from err
-    else:
-        return lz4.block
+        from isal import isal_zlib as zlib
```

Comment:
Oh, that's why it's coming from extras now.

It's a bad idea to choose a library based on what exists on the system—it will work slightly differently from `zlib`, and maybe it will have its own bugs. This would make a bug on one system non-reproducible on another system unless the bug-reporter uses environments correctly and reports the whole environment in which the test was performed.

If we're going to have this capability, it has to be opt-in, within Python so that you can do it in Jupyter (i.e. not based on environment variables). For writing, we can define a new `Compression` subclass with a different name, so that users can say

```python
with uproot.recreate("out.root", compression=uproot.ISAL_ZLIB(3)) as file:
    ...
```

and it would use `isal_zlib` to do the compression and report it as ZLIB in the ROOT file.

If `isal_zlib` is desired for reading, that's more complicated because the decompressor is chosen based on a two-byte sequence before each compressed block. (It has nothing to do with what `fCompress` is set to in the TFile, TTree, or TBranch level.) The only way to inject the user's choice would be to have a global setting,

```python
uproot.ZLIB.is_isal = True
```

Although now that I look at it, it's not a bad interface.

Reply:
Yea I think `uproot.ZLIB.use_isal = True` or what you wrote would be a good way to go.

`use_isal` would be opt and and attempt only that import if set, and error if not available, and would properly affect compression and decompression.